### ISSUE TITLE
add some basic glusto-test scripts

### ISSUE THREAD

**Issue by pkalever** [gluster/gluster-block#41]

**Comment by shwetha-h-p** [comment]
I am working on this issue.

**Comment by sankarshanmukhopadhyay** [comment]
This would need additional detail so that we have on record what the scope started with and whether we have that in place once work is completed.

**Comment by pkalever** [comment]
We wanted to have a basic CI/Jenkins job, which gets triggered on every PR we submit, in-order we catch regressions, in the initial phase.

Initially, we thought of adding some basic bash scripts which we run on Centos CI duffy's. But again requesting multiple duffy's might be a constraint. To solve that we thought of using vagrant to create VM's inside one single duffy.

Since running tests for HA will incur additional management as in, configuring multiple nodes and collecting results from all of them will need some sort of framework. We thought glusto will fit the case, hence decided to start writing tests with glusto suite.

Once we have tests written for gluster-block using glusto suite, then we want do a case study for choosing b/w vagrant vs 'a basic script using libvirt for VM's creation'.

Hopefully by next week we can start running a Centos CI job.

**Comment by sankarshanmukhopadhyay** [comment]
I'll attempt to break this down so as to validate that I understand this plan of action correctly.

- There is a need to "build" on a per PR
- Each of these builds will be deployed
- The deployed builds with be checked for functionality and regressions

is the above the total outcome that is desired?

**Comment by sankarshanmukhopadhyay** [comment]
Alright. I am going to make 2 assumptions here
- the component has unit tests
- the functionality to be tested is well described to have tests added to Glusto library

Does the component also have the attributes which enable easy debugging of test failures viz. descriptive logging and such?

**Comment by pkalever** [comment]
> I'll attempt to break this down so as to validate that I understand this plan of action correctly.
> 
> There is a need to "build" on a per PR
> Each of these builds will be deployed
> The deployed builds with be checked for functionality and regressions
> is the above the total outcome that is desired?

Yes, this is the overall expectation.

We want to start with adding limited test cases for now, once we have the job that gets trigger per PR in place, we can append more tests to them in future.

**Comment by sankarshanmukhopadhyay** [comment]
Alright. I'll wait for @shwetha-h-p to define/scope and document the first set of limited test cases along with @pkalever Once that is in place, it helps ascertain what value we accrue from repeat runs using instrumentation via the Glusto framework. Thanks for taking time to clarify all of this.

**Comment by pkalever** [comment]
> Alright. I am going to make 2 assumptions here
> 
> the component has unit tests

To be honest, we are currently bad here, we have added tests that check for create/delete/list and few other units ...
But we want to improve this in the near future.

> the functionality to be tested is well described to have tests added to Glusto library

Yes

> Does the component also have the attributes which enable easy debugging of test failures viz. descriptive logging and such?

Yes, We have good logging in gluster-block, I believe :-)  But we will definitely work on improving, if required.

**Comment by pkalever** [comment]
@bhumikagoyal This is the issue for glusto based gluster-block tests.

**Comment by pkalever** [comment]
Won't Fix, closing now.
